agent/internal/task_config.go

```diff
@@ -20,54 +20,67 @@ import (
 
 type TaskConfig struct {
 	Distro             *apimodels.DistroView
-	ProjectRef         *model.ProjectRef
-	Project            *model.Project
-	Task               *task.Task
-	BuildVariant       *model.BuildVariant
-	Expansions         *util.Expansions
+	ProjectRef         model.ProjectRef
```

Comment:
Why don't we want these to be pointers anymore? This is a lot more stuff to pass around

Reply:
This avoids an issue where I've noticed a lot of agent code is paranoia nil checking everything because nobody's sure if the pointer indicates they're optional values. In reality, after the task is set up, there's no case where these could be nil and possibly run the task, so they're not optional.

The thing that gets passed around between functions a lot is the `*TaskConfig`, so it's not copying structs very much. For the case of expansions, the `util.Expansions` is already implicitly a pointer type since it's just a wrapper type for a map.